src/vcSettingsUI.cpp

```diff
@@ -21,7 +21,62 @@
 #include "udStringUtil.h"
 #include "udFile.h"
 
-#define MAX_DISPLACEMENT 1000.f
+#define MAX_DISPLACEMENT 10000.f
+
+/*
+Add errors by placing an entry in the language file with the error message, this entry name goes below in settingsErrors[].
+Add a colour in vec4 RGBA format below, and add to the error enum vcSE_ vcSettingsErrors then use vcSettingsUI_Set/Unset/Check Error functions
+*/
+static const char *settingsErrors[] =
+{
+  "bindingsErrorBound",
+  "bindingsSelectKey"
+};
+
+static const ImVec4 settingsErrorColours[] =
+{
+  ImVec4(1, 0, 0, 1),
+  ImVec4(1, 1, 1, 1)
+};
+
+static const vcSettingCategory categoryMapping[] =
+{
+  vcSC_Appearance,
+  vcSC_InputControls,
+  vcSC_Viewport,
+  vcSC_MapsElevation,
+  vcSC_Visualization,
+  vcSC_Bindings,
+  vcSC_Convert,
+  vcSC_Count,
+  vcSC_Count,
+  vcSC_Count,
+  vcSC_Count
+};
+UDCOMPILEASSERT(vcSC_Count == 10, "Update the above mapping if necessary");
+UDCOMPILEASSERT(vcSR_Count == 11, "Update the above mapping if necessary");
```

Comment:
These are very strange asserts and it's very difficult to tell why you've got both, your message doesn't explain anything. Based on counting the number of items in `categoryMapping`, I assume that `vcSR_Count == 11` is supposed to be ensuring that `categoryMapping` has 11 items? Did you consider having `categoryMapping[vcSR_Count]` to enforce the length of the array instead? I assume that `vcSC_Count == 10` is to make sure the values are correct, I would suggest using a switch statement which will make sure that `vcSC_Appearance` is always mapped to `vcSR_Appearance`, or have a number of asserts that ensure that `vcSR_Appearance == 0` and so on. Overall, these asserts aren't making it clear to anyone what is supposed to happen when they fail, and having to decipher meaning is really counter productive.

Reply:
It's to ensure that if a settings category or settings ui region is added that the dev will create an entry if necessary in the mapping table such that the appropriate setting category can be reset or otherwise invoked in relation to the corresponding settings ui region in the settings page, I've documented this better